jax/_src/sharding_impls.py

```diff
@@ -1604,7 +1604,9 @@ def is_single_device_sharding(sharding: sharding.Sharding) -> bool:
   return sharding.num_devices == 1 and not isinstance(sharding, PmapSharding)
 
 def make_key_array_phys_sharding(aval, sharding):
-  if is_single_device_sharding(sharding):
+  if isinstance(sharding, UnspecifiedValue):
```

Comment:
This should not be handling unspecified. Please skip unspecified whereever `physical_sharding` is called i.e. skip in caller not in callee.

Reply:
Done.